app/src/main/java/io/homeassistant/companion/android/background/LocationBroadcastReceiver.kt

```diff
@@ -0,0 +1,144 @@
+package io.homeassistant.companion.android.background
+
+import android.Manifest.permission.ACCESS_COARSE_LOCATION
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.os.BatteryManager
+import android.os.Build
+import android.util.Log
+import androidx.core.app.ActivityCompat
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
+import io.homeassistant.companion.android.DaggerReceiverComponent
+import io.homeassistant.companion.android.common.dagger.Graph
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import io.homeassistant.companion.android.domain.integration.UpdateLocation
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class LocationBroadcastReceiver : BroadcastReceiver() {
+
+    companion object {
+        const val ACTION_REQUEST_LOCATION_UPDATES =
+            "io.homeassistant.companion.android.background.REQUEST_UPDATES"
+        const val ACTION_PROCESS_LOCATION =
+            "io.homeassistant.companion.android.background.PROCESS_UPDATES"
+
+        private const val TAG = "LocBroadcastReceiver"
+    }
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationUseCase
+
+    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    override fun onReceive(context: Context, intent: Intent) {
+        DaggerReceiverComponent.builder()
+            .appComponent(Graph(context).appComponent)
```

Comment:
I will add some better handling around this.  I did this because the API doesn't guarantee that context.applicationContext will return an instance of our Application.  Most times it does, so we can get away with it but I'll handle both cases.